calm_adapter/calm_deletion_checker/src/main/scala/uk/ac/wellcome/platform/calm_deletion_checker/DefectiveChecker.scala

```diff
@@ -65,16 +66,23 @@ trait DefectiveChecker[Item] {
 
 class ApiDeletionChecker(calmApiClient: CalmApiClient)(
   implicit ec: ExecutionContext)
-    extends DefectiveChecker[CalmSourcePayload] {
+    extends DefectiveChecker[CalmSourcePayload]
+    with Logging {
 
   def test(records: Set[CalmSourcePayload]): Future[Int] =
-    calmApiClient
-      .search(
-        records
-          .map(record => CalmQuery.RecordId(record.id))
-          .reduce[CalmQuery](_ or _)
-      )
-      .map {
+    for {
+      session <- calmApiClient
+        .search(
+          records
+            .map(record => CalmQuery.RecordId(record.id))
+            .reduce[CalmQuery](_ or _)
+        )
+      _ <- calmApiClient.abandon(session.cookie).recover {
```

Comment:
Probably worth a comment explaining why we need to Abandon sessions here? (Or think we need to.)